plugins/modules/keycloak_role.py

```diff
@@ -77,6 +77,42 @@
         description:
             - A dict of key/value pairs to set as custom attributes for the role.
             - Values may be single values (e.g. a string) or a list of strings.
+    composite:
+        description:
+            - If C(true), the role is a composition of other realm and/or client role.
+        default: false
+        type: bool
+        version_added: 7.0.0
+    composites:
+        description:
+            - List of roles to include to the composite realm role.
+            - If the composite role is a client role, the C(clientId) (not ID of the client) must be specified.
+        default: []
+        type: list
+        elements: dict
+        version_added: 7.0.0
+        suboptions:
+            name:
+                description:
+                    - Name of the role. This can be the name of a REALM role or a client role.
+                type: str
+                required: true
+            client_id:
+                description:
+                    - Client ID if the role is a client role. Do not include this option for a REALM role.
+                    - Use the client ID you can see in the Keycloak console, not the technical ID of the client.
+                type: str
+                required: false
+                aliases:
+                    - clientId
+            state:
+                description:
+                    - Create the composite if present, remove it if absent.
```

Comment:
How does this state interact with the top-level state parameter of the module? Some feedback from the module maintainers would probably be helpful here, I'm not sure whehter adding this additional complexity of having a state per composite is a good idea.

Reply:
This is a way to remove the role with absent state from the composites and keeping the role with present state. Those state are only useful if the module state parameter is present.
Here is an example:
composites:
  - name: admin
    state: absent
  - name: view-clients
    client_id: test-realm
    state: present
This way we can be sure that admin realm role will be absent from composites but the view-clients for client test-realm will be present.